backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt

```diff
@@ -130,11 +131,49 @@ internal fun emitLLVM(context: Context) {
         irModule.acceptVoid(codegenVisitor)
     }
 
+    when (context.shouldEmbedBitcode()) {
+        EmbedBitcode.ON -> embedBitcode(context, marker = false)
+        EmbedBitcode.MARKER -> embedBitcode(context, marker = true)
+        EmbedBitcode.OFF -> {}
+    }
+
     if (context.shouldContainDebugInfo()) {
         DIFinalize(context.debugInfo.builder)
     }
 }
 
+private fun embedBitcode(context: Context, marker: Boolean) {
+    val family = context.config.platform.configurables.target.family
+    val isApple = family == Family.IOS || family == Family.OSX
+
+    val bytes = if (marker) {
+        byteArrayOf()
+    } else {
+        val buffer = LLVMWriteBitcodeToMemoryBuffer(context.llvmModule)
+        LLVMGetBufferStart(buffer)!!.getBytes(LLVMGetBufferSize(buffer))
+    }
+
+    context.llvm.staticData.placeGlobalArray(
+            name = "konan.embedded.module",
+            elemType = int8Type,
+            elements = bytes.map { Int8(it) },
+            isExported = true
```

Comment:
What's the purpose of passing `isExported = true` and then setting linkage to private?
Passing `isExported = true` just sets linkage to external (i.e. public).